### PR TITLE
docs(subject): Subject documentation for new docs app.

### PR DESCRIPTION
**Pull request by ajcrites**
**Description:**

* Updated subject.md from old docs for consistency with the new API
* Copied and linked this to the new docs app

I'm not sure if we want to remove things that were copied over from `doc`. I'm also not sure whether the migration / update was done properly. I'm happy to address any feedback.

**Related issue (if exists):** #3953 (partial)

### CODE REVIEW

**Comment by pertrai1** [comment]
This looks good. One note is that the newer examples are showing the output in the same code example markdown rather than a newer one below. Example: https://github.com/ReactiveX/rxjs/blob/master/src/internal/observable/forkJoin.ts#L85

File: docs_app/content/guide/subject.md [hunk 1]
```diff
@@ -0,0 +1,364 @@
+# Subject
+
+**What is a Subject?** An RxJS Subject is a special type of Observable that allows values to be multicasted to many Observers. While plain Observables are unicast (each subscribed Observer owns an independent execution of the Observable), Subjects are multicast.
+
+<span class="informal">A Subject is like an Observable, but can multicast to many Observers. Subjects are like EventEmitters: they maintain a registry of many listeners.</span>
```
Comment (pertrai1) [inline docs_app/content/guide/subject.md]: I have wondered what is the purpose of this span class used for?

File: docs_app/content/guide/subject.md [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+  next: (v) => console.log(`observerB: ${v}`)
+});
+
+subject.next(1);
+subject.next(2);
+```
+
+With the following output on the console:
+
+```none
```
Comment (pertrai1) [inline docs_app/content/guide/subject.md]: I would combine this with the above code example just by showing the log below:

```
// Logs:
// observerA: 1
// observerB: 1
// observerA: 2
// observerB: 2
```

File: docs_app/content/guide/subject.md (hunk 1 above)
Comment (pertrai1) [reply docs_app/content/guide/subject.md]: I found it. It is supposed to be used as the blockquote but the new docs does not have the css in place to show it that way.

**Comment by ajcrites** [comment]
Comments addressed. I didn't really know where to put the style. I also changed it to just italicize the text instead of use the cursive font since it seems like Kalam isn't included. The cursive is a little _too_ informal.

File: docs_app/content/navigation.json [hunk 3]
```diff
@@ -24,7 +24,13 @@
 		{
 			"url": "guide/overview",
 			"title": "Overview",
-			"tooltip": "RxJS Overview"
+			"tooltip": "RxJS Overview",
+			"children": [
+				{
+					"url": "guide/subject",
+					"title": "Subjects in RxJS"
```
Comment (JWO719) [inline docs_app/content/navigation.json]: 'Subjects' would be totally fine for the title. I think it is totally clear that it is about rxjs :D

Comment (JWO719) [inline docs_app/content/navigation.json]: 'Subjects' would be totally fine for the title. I think it is totally clear that it is about rxjs :D

File: docs_app/content/guide/subject.md [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+
+subject.next(5);
+subject.complete();
+
+// Logs:
+// observerA: 5
+// observerB: 5
+```
+
+The AsyncSubject is similar to the [`last()`](../class/es6/Observable.js~Observable.html#instance-method-last) operator, in that it waits for the `complete` notification in order to deliver a single value.
```
Comment (JWO719) [inline docs_app/content/guide/subject.md]: I would link to last with: `["last"](/api/operators/last)`

File: docs_app/content/navigation.json (hunk 3 above)
Comment (ajcrites) [reply docs_app/content/navigation.json]: I was just trying to clarify that this was "Subjects, the RxJS construct" vs. "Subjects (e.g. topics) about RxJS)," mostly for newer readers.  I can just change it to "Subjects" though -- that may be clear enough.

**Comment by ajcrites** [comment]
Updated for requested changes.

**Pull request merged**